src/annotator/config/test/url-from-link-tag-test.js

```diff
@@ -0,0 +1,67 @@
+import { urlFromLinkTag } from '../url-from-link-tag';
+
+describe('annotator/config/url-from-link-tag', () => {
+  function appendLink(href, rel, type) {
+    const link = document.createElement('link');
+    link.type = 'application/annotator+' + type;
+    link.rel = rel;
+    if (href) {
+      link.href = href;
+    }
+    document.head.appendChild(link);
+    return link;
+  }
+
+  it('returns the url from the matching link tag of type html', () => {
+    const link = appendLink('http://example.com/app.html', 'sidebar', 'html');
+    assert.equal(
+      urlFromLinkTag(window, 'sidebar', 'html'),
+      'http://example.com/app.html'
+    );
+    link.remove();
+  });
+
+  it('returns the url from the matching link tag of type javascript', () => {
+    const link = appendLink(
+      'http://example.com/app.html',
+      'hypothesis-client',
+      'javascript'
+    );
+    assert.equal(
+      urlFromLinkTag(window, 'hypothesis-client', 'javascript'),
+      'http://example.com/app.html'
+    );
+    link.remove();
+  });
+
+  context('when there are multiple annotator+html notebook links', () => {
+    it('returns the href from the first notebook link found', () => {
+      const link1 = appendLink('http://example.com/app1', 'notebook', 'html');
+      const link2 = appendLink('http://example.com/app2', 'notebook', 'html');
+      assert.equal(
+        urlFromLinkTag(window, 'notebook', 'html'),
+        'http://example.com/app1'
+      );
+      link1.remove();
+      link2.remove();
+    });
+  });
+
+  context('incorrect link tag', () => {
```

Comment:
```suggestion
  context('missing or invalid link tag', () => {
```

The contents of this block encompass both missing `<link>` tags and ones that are invalid because they have no `href`.